src/main/java/io/unityfoundation/auth/HasPermissionRequest.java

```diff
@@ -0,0 +1,13 @@
+package io.unityfoundation.auth;
+
+import io.micronaut.serde.annotation.Serdeable;
+import jakarta.validation.constraints.NotNull;
+import java.util.List;
+
+@Serdeable
+public record HasPermissionRequest(
+    @NotNull Long tenantId,
+    @NotNull Long serviceId,
```

Comment:
@jrw972 @romannaglic Can serviceId be a String instead? I was hoping to derive the value from application.yml's micronaut.application.name configuration (right now it is 'app' but I intend on changing it to "Libre311").

Reply:
We could require that service name be unique and take that in the request.

Reply:
serviceId and tenantId chenged to string.